inductiva/geometry/cad_object.py

```diff
@@ -0,0 +1,236 @@
+of the objects to be used inside our simulators, this class
+provides a simple interface to read the objects from a native
+CAD file.
+"""
+from typing import List, Literal, Optional, Union
+
+try:
+    import pyvista as pv
+except ImportError:
+    pv = None
+
+from inductiva.utils import optional_deps
+
+
+class CADObject:
+    """Class to manipulate CADObjects."""
+
+    @optional_deps.needs_fluids_extra_deps
+    def __init__(self, filename):
+        """"Initialize the CADObject class."""
+
+        self.filename = filename
+        self.data = pv.read(filename)
+
+    def __repr__(self):
+        """Print information about the CAD object.
+        
+        TODO: Add more information about the object.
+        Existing regions, materials, etc...
+        """
+
+        print("\nCADObject details: ")
+        print(self.data)
+        print(f"\nMesh center: {self.data.center}\n")
+
+        return f"CADObject filename: {self.filename}"
+
+    def translate(self, vector: List[float], inplace: bool = False):
+        """Translate the CADObject by a vector.
+
+        Args:
+            vector: The vector to translate the object.
+            inplace: If True, the object is translated in place.
+        """
+        check_spatial_vector(vector, "Vector")
+
+        print("Translate object by: ", vector)
+        data = self.data.translate(vector, inplace=inplace)
+
+        if inplace:
+            self.data = data
+
+        return self
+
+    def to_origin(self, inplace: bool = False):
+        """Translate the CADObject to the origin.
+
+        Args:
+            inplace: If True, the object is translated in place.
+        """
+
+        print("Translating object to the origin.")
+        data = self.data.translate([-value for value in self.data.center])
+
+        if inplace:
+            self.data = data
+
+        return self
+
+    def to_ground(self, inplace: bool = False):
+        """Translate the CADObject to the ground.
+
+        Args:
+            inplace: If True, the object is translated in place.
+        """
+
+        print("Translating object to the ground.")
+        data = self.data.translate([0., 0., -self.data.center[2]])
+
+        if inplace:
+            self.data = data
+
+        return self
+
+    def rotate_around_vector(self,
+                             vector: List[float],
+                             center: Optional[List[float]] = None,
+                             angle: float = 0.,
+                             inplace: bool = False):
+        """Rotate the CADObject by an angle around a vector.
+        
+        Args:
+            vector: 3D vector to rotate the object around.
+            center: 3D center point of the rotation.
+                Default is None, which is converted to [0., 0., 0.]
+            angle: Angle in degrees to rotate the object.
+            inplace: If True, the object rotation is saved in the data.
+        """
```

Comment:
Add a "Returns" section in the docstring to clarify the return type and behavior:
```
CADObject: The CADObject instance after rotation.
```